core/federated/RTI/rti_lib.c

```diff
@@ -53,6 +54,20 @@ int create_server(int32_t specified_port, uint16_t port, socket_type_t socket_ty
     int socket_descriptor = -1;
     if (socket_type == TCP) {
         socket_descriptor = socket(AF_INET, SOCK_STREAM, 0);
+
+        // Disable Nagle algorithm which bundles together small TCP messages to
+        //  reduce network traffic
+        int flag = 1;
+        int result = setsockopt(socket_descriptor,            /* socket affected */
+                                IPPROTO_TCP,     /* set option at TCP level */
+                                TCP_NODELAY,     /* name of option */
```

Comment:
This code is copied three times in this PR. Should we just put this in a library function named `int disable_delayed_transmission(int sockfd)` or something like that?